code/modules/bitrunning/components/avatar_connection.dm

```diff
@@ -101,7 +85,7 @@
 
 	var/mob/living/avatar = parent
 
-	disconnect_avatar_signals()
+	UnregisterFromParent()
```

Comment:
You shouldn't be calling this yourself, and you wouldn't need to if you always qdel in this proc. Is the early return from hosting_netpod being null supposed to stop qdel(src)?

Reply:
yes, it prevents them from returning to their bodies period. the case should never occur, netpods handle being deleted/destroyed etc by signaling this proc. 

i suppose now that this is a component and handles the body transference itself (it used to be the netpod that did this), i can let it fall through and qdel